unittest/gunit/components/percona_telemetry/data_provider-t.cc

```diff
@@ -0,0 +1,439 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#define private public
```

Comment:
Dangerous territory...

Reply:
Not really. It is only for unit test binary. No danger of propagating it into the real telemetry *.so.
This trick is used in other unit tests as well. Allows testing of private methods. Here it is necessary, as we have only one public method which does everything by calling private one by one. Creating unit test for this public method makes no sense, so better to test private methods one by one.